### PR TITLE
Add Java hibernate strategy for Remote workers and update dependencies and logic

### PR DESCRIPTION
**Pull request by mishogv**
Close: https://github.com/SoftUni-Internal/exam-systems-issues/issues/608

### CODE REVIEW

File: OJS.Workers.ExecutionStrategies/Java/JavaSpringAndHibernateProjectExecutionStrategy.cs [hunk 1]
```diff
@@ -501,5 +429,39 @@ private string ExtractEntryPointFromPomXml(string submissionFilePath)
...
+
+            return this.GetMavenErrorsComment(testOutput);
+        }
+
+        private bool ValidateFolderStructure(string submissionFilePath)
+        {
+            var paths = FileHelpers.GetFilePathsFromZip(submissionFilePath).ToList();
+
+            return paths.Any(x => x.StartsWith(MainCodeFolderPattern)) && paths.Any(x => x.StartsWith(PomXmlFileNameAndExtension));
+        }
```
Comment (gogo4ds) [inline OJS.Workers.ExecutionStrategies/Java/JavaSpringAndHibernateProjectExecutionStrategy.cs:465]: Shouldn't `.All` be used here like:
`paths.All(x => x.StartsWith(MainCodeFolderPattern) || x.StartsWith(PomXmlFileNameAndExtension))`.
Do we assume a structure is valid if there are any other paths than these?

File: OJS.Workers.ExecutionStrategies/Java/JavaSpringAndHibernateProjectExecutionStrategy.cs [hunk 2]
```diff
@@ -501,5 +429,39 @@ private string ExtractEntryPointFromPomXml(string submissionFilePath)
...
             return packageName.InnerText.Trim();
         }
+
+        private string GetMavenErrorsComment(string testOutput)
+        {
+            var sb = new StringBuilder();
+
+            foreach (var line in testOutput.Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries)
+                         .Where(x => x
+                             .StartsWith("[ERROR]") || x.StartsWith("[FAILURE]")))
```
Comment (gogo4ds) [inline OJS.Workers.ExecutionStrategies/Java/JavaSpringAndHibernateProjectExecutionStrategy.cs:439]: You have extracted `MavenErrorFailurePattern`, why not reuse it here. Also [FAILURE] could be added to the pattern and used like a regex.

File: OJS.Workers.ExecutionStrategies/Java/JavaSpringAndHibernateProjectExecutionStrategy.cs [hunk 3]
```diff
@@ -152,6 +99,12 @@ protected override string ClassPathArgument
             try
             {
                 submissionFilePath = this.CreateSubmissionFile(executionContext);
+                var isValid = this.ValidateFolderStructure(submissionFilePath);
+
+                if (!isValid)
+                {
+                    throw new ArgumentException("Folder structure is invalid!");
```
Comment (gogo4ds) [inline OJS.Workers.ExecutionStrategies/Java/JavaSpringAndHibernateProjectExecutionStrategy.cs:106]: Can you add more clear message to help the user understand what we are expecting to be a **valid** folder structure.

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: OJS.Workers.ExecutionStrategies/Java/JavaSpringAndHibernateProjectExecutionStrategy.cs (hunk 1 above)
Comment (mishogv) [reply OJS.Workers.ExecutionStrategies/Java/JavaSpringAndHibernateProjectExecutionStrategy.cs:465]: Yes, we are not limiting users to not having different paths. They can add resources or something else in the solution that we do not care about and the whole thing that the validation should do here is to validate that there is a file /pom.xml and that there are files for compiling in the main code folder.

**Review by gogo4ds — APPROVED** [review]

**Pull request merged**